async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -365,6 +394,230 @@ impl<'a> futures::Stream for Stream<'a> {
     }
 }
 
+/// Used for building configuration for a [Stream]. Created by a [stream_builder] on a [Consumer].
+///
+/// # Examples
+///
+/// ```no_run
+/// # #[tokio::main]
+/// # async fn main() -> Result<(), async_nats::Error>  {
+/// use futures::StreamExt;
+/// let client = async_nats::connect("localhost:4222").await?;
+/// let jetstream = async_nats::jetstream::new(client);
+///
+/// let consumer = jetstream
+///     .get_stream("events").await?
+///     .get_consumer("pull").await?;
+///
+/// let mut messages = consumer.stream_builder()
+///     .max_messages_per_batch(100)
+///     .max_bytes_per_batch(1024)
+///     .into_stream().await?;
+///
+/// while let Some(message) = messages.next().await {
+///     let message = message?;
+///     println!("message: {:?}", message);
+///     message.ack().await?;
+/// }
+/// # Ok(())
+/// # }
+pub struct StreamBuilder<'a> {
+    batch: usize,
+    max_bytes: usize,
+    hearbeat: Duration,
+    expires: usize,
+    consumer: &'a Consumer<Config>,
+}
+
+impl<'a> StreamBuilder<'a> {
```

Comment:
Alternatively, since we store the configuration already in Stream.
We can consume it and have these methods act as if they're combinators?

Reply:
Was thinking about that, especially when started writing tests and using it, but I think it's better  to not consumer.

Otherwise Stream builder would act differently than Batch, and we ... might want to allow calling `stream` again after error happening? Maybe, maybe no, but if we consume, its a no for sure.

Reply:
Yeah I'm unsure too, lets start with this and iterate over the next couple of releases.